pkg/image/api/helper.go

```diff
@@ -991,3 +992,57 @@ func SetContainerImageEntrypointAnnotation(annotations map[string]string, contai
 func LabelForStream(stream *ImageStream) string {
 	return fmt.Sprintf("%s/%s", stream.Namespace, stream.Name)
 }
+
+// JoinImageSignatureName joins image name and custom signature name into one string with # separator.
+func JoinImageSignatureName(imageName, signatureName string) (string, error) {
+	if len(imageName) == 0 {
+		return "", fmt.Errorf("imageName may not be empty")
+	}
+	if len(signatureName) == 0 {
+		return "", fmt.Errorf("signatureName may not be empty")
+	}
+	if strings.Count(imageName, "#") > 0 || strings.Count(signatureName, "#") > 0 {
+		return "", fmt.Errorf("neither imageName nor signatureName can contain '#'")
+	}
+	return fmt.Sprintf("%s#%s", imageName, signatureName), nil
+}
+
+// SplitImageSignatureName splits given signature name into image name and signature name.
+func SplitImageSignatureName(imageSignatureName string) (imageName, signatureName string, err error) {
+	segments := strings.SplitN(imageSignatureName, "#", 3)
```

Comment:
We use SplitN because regular split can create an unbounded number of
objects.

On Jul 1, 2016, at 8:32 AM, David Eads <notifications@github.com> wrote:

In pkg/image/api/helper.go
<https://github.com/openshift/origin/pull/9181#discussion_r69317140>:

> +func JoinImageSignatureName(imageName, signatureName string) (string, error) {
> +	if len(imageName) == 0 {
> +		return "", fmt.Errorf("imageName may not be empty")
> +	}
> +	if len(signatureName) == 0 {
> +		return "", fmt.Errorf("signatureName may not be empty")
> +	}
> +	if strings.Count(imageName, "#") > 0 || strings.Count(signatureName, "#") > 0 {
> +		return "", fmt.Errorf("neither imageName nor signatureName can contain '#'")
> +	}
> +	return fmt.Sprintf("%s#%s", imageName, signatureName), nil
> +}
> +
> +// SplitImageSignatureName splits given signature name into image name and signature name.
> +func SplitImageSignatureName(imageSignatureName string) (imageName, signatureName string, err error) {
> +	segments := strings.SplitN(imageSignatureName, "#", 3)

regular split and see what you get

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/9181/files/bbf390c0306766c27a49c808deddd78621151fcc#r69317140>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p4yLqIeI27FV0wxRi3nDWQTb4Gywks5qRTLkgaJpZM4IvEFx>
.